rust/lance/src/index/vector/ivf/io.rs

```diff
@@ -446,12 +446,12 @@ async fn build_hnsw_quantization_partition(
     let mut metric_type = metric_type;
     if metric_type == MetricType::Cosine {
         // Normalize vectors for cosine similarity
-        vectors = normalize_arrow(&vectors)?;
+        vectors =
+            Arc::new(spawn_cpu(move || Ok(normalize_fsl(vectors.as_fixed_size_list())?)).await?);
```

Comment:
How big is this `vectos` usually? is it worth to create a new thread?

Reply:
it's all the vectors of one partition, could be very big for large dataset